### ISSUE TITLE
Test failures

### ISSUE THREAD

**Issue by fabaff** [jupyterhub/oauthenticator#432]
### Bug description
Not all tests are passing with  Python 3.8.9 and `pytest-6.2.3`. This happens during the build process for NixOS packages.

```bash
============================= test session starts ==============================
platform linux -- Python 3.8.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /build/oauthenticator-14.0.0
plugins: asyncio-0.14.0, requests-mock-1.8.0
collected 58 items                                                             

oauthenticator/tests/test_auth0.py .                                     [  1%]
oauthenticator/tests/test_azuread.py .FFFF                               [ 10%]
oauthenticator/tests/test_bitbucket.py ...                               [ 15%]
oauthenticator/tests/test_cilogon.py .....                               [ 24%]
oauthenticator/tests/test_generic.py ........                            [ 37%]
oauthenticator/tests/test_github.py ...                                  [ 43%]
oauthenticator/tests/test_gitlab.py ....                                 [ 50%]
oauthenticator/tests/test_globus.py .........                            [ 65%]
oauthenticator/tests/test_google.py ........                             [ 79%]
oauthenticator/tests/test_mediawiki.py F.                                [ 82%]
oauthenticator/tests/test_oauth2.py ..                                   [ 86%]
oauthenticator/tests/test_okpy.py ..                                     [ 89%]
oauthenticator/tests/test_openshift.py ......                            [100%]

=================================== FAILURES ===================================
```

<details>

```bash
______________________________ test_azuread[None] ______________________________

username_claim = None
azure_client = <oauthenticator.tests.mocks.MockAsyncHTTPClient object at 0x7ffff4c40fd0>

    @pytest.mark.parametrize(
        'username_claim',
        [
            None,
            'name',
            'oid',
            'preferred_username',
        ],
    )
    async def test_azuread(username_claim, azure_client):
        authenticator = AzureAdOAuthenticator(
            tenant_id=str(uuid.uuid1()),
            client_id=str(uuid.uuid1()),
            client_secret=str(uuid.uuid1()),
        )
        if username_claim:
            authenticator.username_claim = username_claim

        handler = azure_client.handler_for_user(
>           user_model(
                tenant_id=authenticator.tenant_id,
                client_id=authenticator.client_id,
                name="somebody",
            )
        )

oauthenticator/tests/test_azuread.py:82:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

tenant_id = '97e4a70c-ae58-11eb-b130-0fc19347ba48'
client_id = '97e4c358-ae58-11eb-ba72-0fc19347ba48', name = 'somebody'

    def user_model(tenant_id, client_id, name):
        """Return a user model"""
        # model derived from https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#v20
        now = int(time.time())
>       id_token = jwt.encode(
            {
                "ver": "2.0",
                "iss": f"https://login.microsoftonline.com/{tenant_id}/v2.0",
                "sub": "AAAAAAAAAAAAAAAAAAAAAIkzqFVrSaSaFHy782bbtaQ",
                "aud": client_id,
                "exp": now + 3600,
                "iat": now,
                "nbf": now,
                "name": name,
                "preferred_username": name,
                "oid": str(uuid.uuid1()),
                "tid": tenant_id,
                "nonce": "123523",
                "aio": "Df2UVXL1ix!lMCWMSOJBcFatzcGfvFGhjKv8q5g0x732dR5MB5BisvGQO7YWByjd8iQDLq!eGbIDakyp5mnOrcdqHeYSnltepQmRp6AIZ8jY",
            },
            os.urandom(5),
        ).decode("ascii")
E       AttributeError: 'str' object has no attribute 'decode'

oauthenticator/tests/test_azuread.py:27: AttributeError
______________________________ test_azuread[name] ______________________________

username_claim = 'name'
azure_client = <oauthenticator.tests.mocks.MockAsyncHTTPClient object at 0x7ffff4c93040>

    @pytest.mark.parametrize(
        'username_claim',
        [
            None,
            'name',
            'oid',
            'preferred_username',
        ],
    )
    async def test_azuread(username_claim, azure_client):
        authenticator = AzureAdOAuthenticator(
            tenant_id=str(uuid.uuid1()),
            client_id=str(uuid.uuid1()),
            client_secret=str(uuid.uuid1()),
        )
        if username_claim:
            authenticator.username_claim = username_claim

        handler = azure_client.handler_for_user(
>           user_model(
                tenant_id=authenticator.tenant_id,
                client_id=authenticator.client_id,
                name="somebody",
            )
        )

oauthenticator/tests/test_azuread.py:82:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

tenant_id = '97f6850d-ae58-11eb-bf2a-0fc19347ba48'
client_id = '97f68629-ae58-11eb-95b8-0fc19347ba48', name = 'somebody'

    def user_model(tenant_id, client_id, name):
        """Return a user model"""
        # model derived from https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#v20
        now = int(time.time())
>       id_token = jwt.encode(
            {
                "ver": "2.0",
                "iss": f"https://login.microsoftonline.com/{tenant_id}/v2.0",
                "sub": "AAAAAAAAAAAAAAAAAAAAAIkzqFVrSaSaFHy782bbtaQ",
                "aud": client_id,
                "exp": now + 3600,
                "iat": now,
                "nbf": now,
                "name": name,
                "preferred_username": name,
                "oid": str(uuid.uuid1()),
                "tid": tenant_id,
                "nonce": "123523",
                "aio": "Df2UVXL1ix!lMCWMSOJBcFatzcGfvFGhjKv8q5g0x732dR5MB5BisvGQO7YWByjd8iQDLq!eGbIDakyp5mnOrcdqHeYSnltepQmRp6AIZ8jY",
            },
            os.urandom(5),
        ).decode("ascii")
E       AttributeError: 'str' object has no attribute 'decode'

oauthenticator/tests/test_azuread.py:27: AttributeError
______________________________ test_azuread[oid] _______________________________

username_claim = 'oid'
azure_client = <oauthenticator.tests.mocks.MockAsyncHTTPClient object at 0x7ffff4b50250>

    @pytest.mark.parametrize(
        'username_claim',
        [
            None,
            'name',
            'oid',
            'preferred_username',
        ],
    )
    async def test_azuread(username_claim, azure_client):
        authenticator = AzureAdOAuthenticator(
            tenant_id=str(uuid.uuid1()),
            client_id=str(uuid.uuid1()),
            client_secret=str(uuid.uuid1()),
        )
        if username_claim:
            authenticator.username_claim = username_claim

        handler = azure_client.handler_for_user(
>           user_model(
                tenant_id=authenticator.tenant_id,
                client_id=authenticator.client_id,
                name="somebody",
            )
        )

oauthenticator/tests/test_azuread.py:82:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

tenant_id = '97f891d9-ae58-11eb-a61d-0fc19347ba48'
client_id = '97f892f7-ae58-11eb-81ea-0fc19347ba48', name = 'somebody'

    def user_model(tenant_id, client_id, name):
        """Return a user model"""
        # model derived from https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#v20
        now = int(time.time())
>       id_token = jwt.encode(
            {
                "ver": "2.0",
                "iss": f"https://login.microsoftonline.com/{tenant_id}/v2.0",
                "sub": "AAAAAAAAAAAAAAAAAAAAAIkzqFVrSaSaFHy782bbtaQ",
                "aud": client_id,
                "exp": now + 3600,
                "iat": now,
                "nbf": now,
                "name": name,
                "preferred_username": name,
                "oid": str(uuid.uuid1()),
                "tid": tenant_id,
                "nonce": "123523",
                "aio": "Df2UVXL1ix!lMCWMSOJBcFatzcGfvFGhjKv8q5g0x732dR5MB5BisvGQO7YWByjd8iQDLq!eGbIDakyp5mnOrcdqHeYSnltepQmRp6AIZ8jY",
            },
            os.urandom(5),
        ).decode("ascii")
E       AttributeError: 'str' object has no attribute 'decode'

oauthenticator/tests/test_azuread.py:27: AttributeError
_______________________ test_azuread[preferred_username] _______________________

username_claim = 'preferred_username'
azure_client = <oauthenticator.tests.mocks.MockAsyncHTTPClient object at 0x7ffff4c39eb0>

    @pytest.mark.parametrize(
        'username_claim',
        [
            None,
            'name',
            'oid',
            'preferred_username',
        ],
    )
    async def test_azuread(username_claim, azure_client):
        authenticator = AzureAdOAuthenticator(
            tenant_id=str(uuid.uuid1()),
            client_id=str(uuid.uuid1()),
            client_secret=str(uuid.uuid1()),
        )
        if username_claim:
            authenticator.username_claim = username_claim

        handler = azure_client.handler_for_user(
>           user_model(
                tenant_id=authenticator.tenant_id,
                client_id=authenticator.client_id,
                name="somebody",
            )
        )

oauthenticator/tests/test_azuread.py:82:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

tenant_id = '97fab67a-ae58-11eb-a74a-0fc19347ba48'
client_id = '97fab849-ae58-11eb-b693-0fc19347ba48', name = 'somebody'

    def user_model(tenant_id, client_id, name):
        """Return a user model"""
        # model derived from https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens#v20
        now = int(time.time())
>       id_token = jwt.encode(
            {
                "ver": "2.0",
                "iss": f"https://login.microsoftonline.com/{tenant_id}/v2.0",
                "sub": "AAAAAAAAAAAAAAAAAAAAAIkzqFVrSaSaFHy782bbtaQ",
                "aud": client_id,
                "exp": now + 3600,
                "iat": now,
                "nbf": now,
                "name": name,
                "preferred_username": name,
                "oid": str(uuid.uuid1()),
                "tid": tenant_id,
                "nonce": "123523",
                "aio": "Df2UVXL1ix!lMCWMSOJBcFatzcGfvFGhjKv8q5g0x732dR5MB5BisvGQO7YWByjd8iQDLq!eGbIDakyp5mnOrcdqHeYSnltepQmRp6AIZ8jY",
            },
            os.urandom(5),
        ).decode("ascii")
E       AttributeError: 'str' object has no attribute 'decode'

oauthenticator/tests/test_azuread.py:27: AttributeError
________________________________ test_mediawiki ________________________________

mediawiki = <requests_mock.mocker.Mocker object at 0x7ffff4a010a0>

    async def test_mediawiki(mediawiki):
        authenticator = new_authenticator()
        handler = Mock(spec=web.RequestHandler,
            get_secure_cookie=Mock(
                return_value=json.dumps(
                    ['key', 'secret']
                )
            ),
            request=Mock(
                query='oauth_token=key&oauth_verifier=me'
            )
        )
>       user = await authenticator.authenticate(handler, None)

oauthenticator/tests/test_mediawiki.py:64:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
oauthenticator/mediawiki.py:123: in authenticate
    identity = await wrap_future(self.executor.submit(handshaker.identify, access_token))
/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/concurrent/futures/thread.py:57: in run
    result = self.fn(*self.args, **self.kwargs)
/nix/store/gnmzzikxgahmbfmgiw7jrlbnfwy77rv2-python3.8-mwoauth-0.3.7/lib/python3.8/site-packages/mwoauth/handshaker.py:126: in identify
    return identify(self.mw_uri, self.consumer_token, access_token,
/nix/store/gnmzzikxgahmbfmgiw7jrlbnfwy77rv2-python3.8-mwoauth-0.3.7/lib/python3.8/site-packages/mwoauth/functions.py:229: in identify
    r = requests.post(url=mw_uri,
/nix/store/534d0l78q3apma8naqgfkxf2vb1p7i0x-python3.8-requests-2.25.1/lib/python3.8/site-packages/requests/api.py:119: in post
    return request('post', url, data=data, json=json, **kwargs)
/nix/store/534d0l78q3apma8naqgfkxf2vb1p7i0x-python3.8-requests-2.25.1/lib/python3.8/site-packages/requests/api.py:61: in request
    return session.request(method=method, url=url, **kwargs)
/nix/store/534d0l78q3apma8naqgfkxf2vb1p7i0x-python3.8-requests-2.25.1/lib/python3.8/site-packages/requests/sessions.py:542: in request
    resp = self.send(prep, **send_kwargs)
/nix/store/m23ylnhcgg56wrvywya9lwyzmydxpz39-python3.8-requests-mock-1.8.0/lib/python3.8/site-packages/requests_mock/mocker.py:131: in _fake_send
    return _original_send(session, request, **kwargs)
/nix/store/534d0l78q3apma8naqgfkxf2vb1p7i0x-python3.8-requests-2.25.1/lib/python3.8/site-packages/requests/sessions.py:655: in send
    r = adapter.send(request, **kwargs)
/nix/store/m23ylnhcgg56wrvywya9lwyzmydxpz39-python3.8-requests-mock-1.8.0/lib/python3.8/site-packages/requests_mock/adapter.py:247: in send
    resp = matcher(request)
/nix/store/m23ylnhcgg56wrvywya9lwyzmydxpz39-python3.8-requests-mock-1.8.0/lib/python3.8/site-packages/requests_mock/adapter.py:227: in __call__
    return response_matcher.get_response(request)
/nix/store/m23ylnhcgg56wrvywya9lwyzmydxpz39-python3.8-requests-mock-1.8.0/lib/python3.8/site-packages/requests_mock/response.py:254: in get_response
    return create_response(request,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

request = <requests_mock.request._RequestObjectProxy object at 0x7ffff49e5be0>
kwargs = {'cookies': <CookieJar[]>, 'reason': None, 'status_code': 200}
connection = <requests_mock.response._FakeConnection object at 0x7ffff4a01d30>
raw = None, body = None
content = 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6Indhc2giLCJhdWQiOiJjbGllbnRfaWQiLCJpc3MiOiJodHRwczovL21ldGEud2lr...DI5ODExMC4wNzIwMzQ0LCJub25jZSI6IjcwMTIxODkzODIxNTIxMTkwMzgxNjIwMjk4MTEwIn0.wD-k1arHBtzIzLm000jZPeTcaWgG1fKL8OEUT-UVvrk'
text = None, json = None, headers = {}

    def create_response(request, **kwargs):
        """
        :param int status_code: The status code to return upon a successful
            match. Defaults to 200.
        :param HTTPResponse raw: A HTTPResponse object to return upon a
            successful match.
        :param io.IOBase body: An IO object with a read() method that can
            return a body on successful match.
        :param bytes content: A byte string to return upon a successful match.
        :param unicode text: A text string to return upon a successful match.
        :param object json: A python object to be converted to a JSON string
            and returned upon a successful match.
        :param dict headers: A dictionary object containing headers that are
            returned upon a successful match.
        :param CookieJar cookies: A cookie jar with cookies to set on the
            response.
        """
        connection = kwargs.pop('connection', _FakeConnection())

        _check_body_arguments(**kwargs)

        raw = kwargs.pop('raw', None)
        body = kwargs.pop('body', None)
        content = kwargs.pop('content', None)
        text = kwargs.pop('text', None)
        json = kwargs.pop('json', None)
        headers = kwargs.pop('headers', {})
        encoding = None

        if content is not None and not isinstance(content, six.binary_type):
>           raise TypeError('Content should be binary data')
E           TypeError: Content should be binary data

/nix/store/m23ylnhcgg56wrvywya9lwyzmydxpz39-python3.8-requests-mock-1.8.0/lib/python3.8/site-packages/requests_mock/response.py:161: TypeError
```
</details>

#### Expected behaviour
All tests passing

#### Actual behaviour
5 tests are failing.

**Comment by minrk** [comment]
I suspect this is a pyjwt version issue. Can you include the output of `pip list`?

**Comment by consideRatio** [comment]
I'm quite confident this was related to pyjwt version issues. This should be resolved with #581 where this package provides the required versions for function going onwards etc.
